src/NServiceBus.Persistence.AzureTable/Subscriptions/AzureSubscriptionStorage.cs

```diff
@@ -160,17 +153,23 @@ public Task<IEnumerable<Subscriber>> GetSubscriberAddressesForMessage(IEnumerabl
         async Task<IEnumerable<Subscriber>> GetSubscriptions(IEnumerable<MessageType> messageTypes, CancellationToken cancellationToken)
         {
             var subscribers = new HashSet<Subscriber>(SubscriberComparer.Instance);
-            var table = client.GetTableReference(subscriptionTableName);
+            var table = client.GetTableClient(subscriptionTableName);
 
             foreach (var messageType in messageTypes)
             {
+                var subscriptionsForMessageType = new List<Subscription>();
                 var name = messageType.TypeName;
-                var lowerBound = TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.GreaterThanOrEqual, name);
-                var upperBound = TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.LessThan, GetUpperBound(name));
-                var query = new TableQuery<Subscription>().Where(TableQuery.CombineFilters(lowerBound, "and", upperBound));
 
-                var subscriptions = await table.ExecuteQueryAsync(query, cancellationToken: cancellationToken).ConfigureAwait(false);
-                var results = subscriptions.Select(s => new Subscriber(DecodeFrom64(s.RowKey), s.EndpointName));
+                var lowerBound = $"PartitionKey ge {name}";
+                var upperBound = $"PartitionKey lt {GetUpperBound(name)}";
+                var query = $"{lowerBound} and {upperBound}";
+
+                var queryResults = table.QueryAsync<Subscription>(query, int.MaxValue, cancellationToken: cancellationToken);
+                await foreach (Page<Subscription> page in queryResults.AsPages().WithCancellation(cancellationToken))
```

Comment:
Since we want everything could we use ToListAsync similar to here https://github.com/Particular/NServiceBus.AzureStorageQueues/pull/895/files#diff-d0bcf593a79e65e5dded2ba4a7f3e4f8eaf6d61452f9aefc62afa708c8fe017aR129?